### PR TITLE
Print str arrays

### PR DESCRIPTION
**Pull request by YersiniaP**

### CODE REVIEW

File: src/libopal.c [hunk 1]
```diff
@@ -2392,6 +2392,49 @@ short print_asm_code(asm_cmd_e cmd_list[], FILE *dest_fp)
...
 
   /// Create strings and their lengths
+  for (int i = 0; i < strs_len; i++)
+    {
+      fprintf (dest_fp, "msg%d: DB \"", i);
+      /// Read each string character
+      for (int j = 0; j < strlen(strs[i])+1; j++)
+        {
+           ///print ASCII values for carriage returns
+           if (strs[i][j] == 13 && strs[i][j+1] == 10)
```
Comment (kedar-damle) [inline src/libopal.c:2402]: This code is trying to replace `CR&NL` character which shouldn't exist inside strings -- unless [get_string_literal_lexeme()](https://github.com/mckerracher/OPaL/blob/23e919ed12caf619242eb0c6cdd22b4d4e97d5a4/src/libopal.c#L902) has a bug, where it should error out.

In this function the sequence `\n` in a string should be replaced by `", 13, 10, "`

Eg. If the program has string like this:
```
1. Hello world\n
2. Operation: \n0. Exit\n1. Add\n2. Subtract\n3. Multiply\n4. Divide\n5. Remainder\n
```
Should result in:
```
1. msg0: DB "Hello world", 13, 10, NULL
2. msg1: DB "Operation: ", 13, 10, "0. Exit", 13, 10, "1. Add", 13, 10, "2. Subtract", 13, 10, "3. Multiply", 13, 10, "4. Divide", 13, 10, "5. Remainder", 13, 10, "", NULL
```

Notice in the second example above, the `\n` after Remainder needs to be followed by an empty `""` else NASM complains.

File: include/libopal.h [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/// @file opal.h
+/// @file libopal.h
```
Comment (kedar-damle) [inline include/libopal.h:1]: :+1:

File: src/libopal.c [hunk 3]
```diff
@@ -2402,7 +2445,7 @@ short print_asm_code(asm_cmd_e cmd_list[], FILE *dest_fp)
  * @brief Print assembly code list to HTML report file
  *
  * @param       cmd_list    Assembly command list to print
- * @param       report_fp   Destination HTML reportfile pointer
+ * @param       dest_fp   Destination HTML reportfile pointer
```
Comment (kedar-damle) [inline src/libopal.c:2448]: :+1:

**Review by kedar-damle — CHANGES_REQUESTED** [review]

File: src/libopal.c (hunk 1 above)
Comment (kedar-damle) [reply src/libopal.c:2402]: I filed the bug #98 for get_string_literal_lexeme() which I noticed while reviewing this code :)
I have a pull request open to fix it.

Comment (YersiniaP) [reply src/libopal.c]: That  makes things much easier :)

File: src/libopal.c [hunk 4]
```diff
@@ -2392,6 +2392,42 @@ short print_asm_code(asm_cmd_e cmd_list[], FILE *dest_fp)
   /// Copy NASM footer file to dest_fp
 
   /// Create strings and their lengths
+  for (int i = 0; i < strs_len; i++)
+    {
+      fprintf (dest_fp, "msg%d: DB \"", i);
+      /// Read each string character
+      for (int j = 0; j < strlen(strs[i])+1; j++)
```
Comment (kedar-damle) [inline src/libopal.c:2399]: You shouldn't have to go 1 past the strlen .. are you seeing issues without it ?

File: src/libopal.c [hunk 5]
```diff
@@ -2392,6 +2392,42 @@ short print_asm_code(asm_cmd_e cmd_list[], FILE *dest_fp)
...
+        {
+           ///print ASCII values for newlines
+           if (strs[i][j] == 10)
```
Comment (kedar-damle) [inline src/libopal.c:2402]: You are again checking if the stream has a newline .. we want to check if user wants to print a newline in the resulting binary.

```
if (strs[i][j] == '\\' && strs[i][j + 1] == 'n')
  {
    fprintf (dest_fp, "\", 13, 10, \"");
    j = j + 1;
  }
```

**Review by kedar-damle — CHANGES_REQUESTED** [review]

File: src/libopal.c (hunk 4 above)
Comment (YersiniaP) [reply src/libopal.c:2399]: No, I might have before my previous fixes, but you're right that it works with it now.

File: src/libopal.c (hunk 5 above)
Comment (YersiniaP) [reply src/libopal.c:2402]: We are going to pretend it is my second covid dose that caused this confusion, and not the fact that I misunderstood this the whole time ;)

File: src/libopal.c [hunk 6]
```diff
@@ -2392,6 +2392,45 @@ short print_asm_code(asm_cmd_e cmd_list[], FILE *dest_fp)
...
+               j = j+ 1;
+             }
+
+           /// directly print all other characters
+           else
+             fprintf (dest_fp, "%c", strs[i][j]);
+        }
+
+      /// NULL terminate string
+      fprintf (dest_fp, ", NULL\n");
```
Comment (kedar-damle) [inline src/libopal.c:2414]: You need to cap the other end of the string with a matching `"` that starts just before the `for` loop .. 

```
fprintf (dest_fp, "\", NULL\n");
```

**Review by kedar-damle — CHANGES_REQUESTED** [review]

File: src/libopal.c (hunk 6 above)
Comment (YersiniaP) [reply src/libopal.c:2414]: Sorry I thought I fixed that, I've been pasting this into a replit to test and apparently I only fixed it there.

**Review by kedar-damle — APPROVED** [review]

**Pull request merged**